lib/src/components/phone_input/countries_dialog.dart

```diff
@@ -84,40 +85,139 @@ class _CountriesDialogState<T> extends State<CountriesDialog<T>> {
   }
 }
 
-class _CountriesList<T> extends StatelessWidget {
+class _CountriesList extends StatefulWidget {
   const _CountriesList({
     required this.items,
     this.zeta,
   });
 
   final Zeta? zeta;
-  final List<DropdownMenuItem<T>> items;
+  final List<CountriesMenuItem> items;
+
+  @override
+  State<_CountriesList> createState() => _CountriesListState();
+
+  @override
+  void debugFillProperties(DiagnosticPropertiesBuilder properties) {
+    super.debugFillProperties(properties);
+    properties.add(IterableProperty<CountriesMenuItem>('items', items));
+  }
+}
+
+class _CountriesListState extends State<_CountriesList> {
+  late final bool _enableSearch = widget.items.length > 20;
+  final _controller = TextEditingController();
+  List<CountriesMenuItem> _items = [];
+
+  @override
+  void initState() {
+    super.initState();
+    _items = List.from(widget.items);
+  }
+
+  void _search(String value) {
+    setState(() {
+      _items = widget.items.where((item) {
+        return item.value.name.toLowerCase().contains(value.toLowerCase()) ||
+            (RegExp(r'^\d+$').hasMatch(value) && item.value.dialCode.indexOf('+$value') == 0);
+      }).toList();
+    });
+  }
+
+  void _clearSearch() {
+    _controller.clear();
+    setState(() {
+      _items = List.from(widget.items);
+    });
+  }
+
+  @override
+  void dispose() {
+    _controller.dispose();
+    super.dispose();
+  }
 
   @override
   Widget build(BuildContext context) {
-    final zeta = this.zeta ?? Zeta.of(context);
+    final zeta = widget.zeta ?? Zeta.of(context);
 
     return AlertDialog(
       surfaceTintColor: zeta.colors.surfacePrimary,
       shape: const RoundedRectangleBorder(borderRadius: ZetaRadius.large),
       content: SizedBox(
         width: double.maxFinite,
-        child: ListView.builder(
-          shrinkWrap: true,
-          itemCount: items.length,
-          itemBuilder: (_, index) => InkWell(
-            onTap: () {
-              Navigator.of(context).pop(items[index].value);
-            },
-            child: Padding(
-              padding: const EdgeInsets.symmetric(
-                vertical: ZetaSpacing.xs,
+        child: Column(
+          mainAxisSize: MainAxisSize.min,
+          children: [
+            if (_enableSearch)
+              Padding(
+                padding: const EdgeInsets.only(bottom: ZetaSpacing.b),
+                child: TextField(
+                  controller: _controller,
+                  onChanged: _search,
+                  decoration: InputDecoration(
+                    hintText: 'Search',
```

Comment:
this hint should be passed in as a property, with search as the default

Reply:
If you don't mind, I plan to completely remove the `TextField`, and replace it with `ZetaSearchBar`, once approved and merged. Then it will become only this:

```
              child: ZetaSearchBar(
                onChanged: (value) {...},
              ),
```

Reply:
Yes I realized this after, I reviewed them in the wrong order!